### DISCUSSION TITLE
Identifying users for communication

### DISCUSSION THREAD

**Discussion by Peterculazh** [lxzan/gws#105] [Q&A] [answered]
Hello, I'm new with Golang but still want to give a try to it for pet project
So my goal is to make a game. For that I need to allow connect multiple users by websocket, store for each user it's own context/session (for storing score and details) and on each frame send for each user it's own data that related to that user.

I did logic for storing score and details into extra struct, but I can't figure out:
1. How I can understand from which user I got a message?
2. How to switch between user's context/session (to make sure with which user I'm gonna to communicate) and send message to that user?

**Comment by lxzan** [comment, marked as answer]
see chatroom example
